requests/utils.py

```diff
@@ -732,6 +732,22 @@ def to_native_string(string, encoding='ascii'):
 
     return out
 
+def validate_header(header):
+    """Verifies that header value doesn't contain leading whitespace or
+    return characters. This prevents unintended header injection.
+
+    header is a tuple, in the format (name, value)
+    """
+    name, value = header
+
+    if isinstance(value, bytes):
+        pat = re.compile(b'^\S[^\\r\\n]*$|^$')
+    else:
+        pat = re.compile(r'^\S[^\r\n]*$|^$')
```

Comment:
That's what I was thinking too (with performance), I just wasn't sure where to declare them. You're suggesting throwing something like this at the top of `utils.py`, yeah?

```
CLEAN_HEADER_REGEX_BYTES = re.compile(b'^\S[^\\r\\n]*$|^$')
CLEAN_HEADER_REGEX_STR = re.compile(r'^\S[^\r\n]*$|^$')
```